pkg/apis/allocation/v1/gameserverallocation.go

```diff
@@ -144,5 +265,10 @@ func (gsa *GameServerAllocation) Validate() ([]metav1.StatusCause, bool) {
 			Message: fmt.Sprintf("Invalid value: %s, value must be either Packed or Distributed", gsa.Spec.Scheduling)})
 	}
 
+	causes, _ = gsa.Spec.Required.Validate("spec.required", causes)
```

Comment:
This will overwrite causes if it's non empty. Both here and below you need to append to causes (like is done on line 263). 

(update: re-reading the validate function, I can see why this works, but it's really confusing to read. I think it's simpler to have validate not take causes and have this function do the appending).

Reply:
I agree! I will make this change! Same as comment below 👍🏻